cai_causal_graph/time_series_causal_graph.py

```diff
@@ -1279,23 +1283,49 @@ def to_numpy_by_lag(self) -> Tuple[Dict[int, numpy.ndarray], List[str]]:
         assert graph.variables is not None
         return adjacency_matrices, graph.variables
 
+    def cache_nodes(self):
+        """Populate the caches for efficient lookup."""
+        self.lag_to_nodes = {}
+        self.variable_name_to_nodes = {}
+        for node in self.get_nodes():
+            # Update lag-to-nodes cache
+            if node.time_lag not in self.lag_to_nodes:
+                self.lag_to_nodes[node.time_lag] = []
+            self.lag_to_nodes[node.time_lag].append(node)
+
+            # Update variable-name-to-nodes cache
+            if node.variable_name not in self.variable_name_to_nodes:
+                self.variable_name_to_nodes[node.variable_name] = []
+            self.variable_name_to_nodes[node.variable_name].append(node)
```

Comment:
based on your implementation of `update_cache`, why not reuse that here? so it would be:
```
for node in self.get_nodes():
   self.update_cache(node)

return self.lag_to_nodes, self.variable_name_to_nodes
```